Battery

tkdesai13 opened the pull request.

fedeoli commented:
Hi Tushar! Do you need to merge into the master for a specific reason or can we keep using this branch for development?

tkdesai13 commented:
> Hi Tushar! Do you need to merge into the master for a specific reason or can we keep using this branch for development?

I think we better keep using the battery branch for the moment and in future, if needed we will merge.

fedeoli committed:
chore: set randflag

fedeoli committed:
del: remove former battery model

fedeoli committed:
feat: LTI unstable test control model

fedeoli committed:
chore: sim setup

fedeoli committed:
feat: plot manipulator and model update

fedeoli committed:
feat: random around zero init

fedeoli committed:
feat: test on LTI unstable model

fedeoli committed:
feat: more complex conrol law (drive)

fedeoli committed:
chore: sim setup

fedeoli committed:
feat: use y in control

fedeoli committed:
chore: update control test model

fedeoli committed:
chore: update params sim

fedeoli committed:
feat: test controller - TODO fix obs update

tkdesai13 committed:
Create ECM.slx

tkdesai13 committed:
Create Battery_simscape_model.slx

tkdesai13 committed:
Create 11_05_2015_SP20-2_DST_50SOC.xls

tkdesai13 committed:
Create ECM_parameters.mat

tkdesai13 committed:
Create 11_05_2015_SP20-2_DST_80SOC.xls

tkdesai13 committed:
Create DST_Current_input_SP20-2_DST_50SOC.xlsx

tkdesai13 committed:
Create SP2_25C_DST_50.xlsx

fedeoli committed:
feat: update control law

fedeoli committed:
feat: measure diff for estimate and control + reference

fedeoli committed:
feat: differentiate estimation and control models

fedeoli committed:
feat: add models for reference and target (MRAC)

fedeoli committed:
chore: rearrange parameters + add history vars

fedeoli committed:
fix: test obs without changing the calls object (handle)

fedeoli committed:
feat: differentiate combined estimation with general sim + delete test old version

fedeoli committed:
chore: plot after test obs

fedeoli committed:
feat: different control laws for ID and control

fedeoli committed:
feat: different filters for ID and control

fedeoli committed:
chore: edit ref and model measure for control purposes

fedeoli committed:
chore: edit model control test

fedeoli committed:
feat: ID model + CTRL model

fedeoli committed:
chore: edit init and reference model

fedeoli committed:
chore: edit combined sim ID + ctrl

fedeoli committed:
chore: edit test ctrl law script and plot

fedeoli committed:
chore: sim setup

fedeoli committed:
feat: globas search

fedeoli committed:
fix: init condition error

fedeoli committed:
feat: estimation model with different structure (no model matching)

fedeoli committed:
chore: params setup

fedeoli committed:
chore: params update

fedeoli committed:
feat: add flags to opt and combined estimation

fedeoli committed:
feat: routh criterion and model analysis

tkdesai13 committed:
Battery ECM dynamics added, still optimisation is diverging

fedeoli committed:
chore: testing new scheme = est + control

fedeoli committed:
chore: remove simulink cache files

fedeoli committed:
chore: update params

fedeoli committed:
feat: add filter on measure (reference.TF filter)

fedeoli committed:
fix: params dynamics in model, no update as it's a algebraic dependence on Z

fedeoli committed:
feat: update with control version of obsopt

fedeoli committed:
chore: gitignore

fedeoli committed:
chore: add simulations results (report on overleaf)

fedeoli committed:
feat: obsopt multistart, remove control design, terminal cost, max opt time, barrier

fedeoli committed:
feat: control for both id and ref track

fedeoli committed:
chore: params and measurements

fedeoli committed:
feat: weight terms post normalisation

fedeoli committed:
chore: model and params in simulation

fedeoli committed:
feat: test model

fedeoli committed:
doc: paper

fedeoli committed:
data: workspace (fedeoli version)

fedeoli committed:
chore: rewrite control with switch case

fedeoli committed:
chore: add 2dim filter (option)

fedeoli committed:
chore: add second measure (option)

fedeoli committed:
chore: change params init model

fedeoli committed:
feat: opt problem to find first vals for params

fedeoli committed:
feat: script to generate data (fedeoli version)

fedeoli committed:
feat: CT model

fedeoli committed:
feat: add reference model

fedeoli committed:
feat: ORC sym analysis (several models)

fedeoli committed:
feat: add true params values in init procedure

fedeoli committed:
feat: add update for all the params

fedeoli committed:
chore: change params and add first_guess

fedeoli committed:
feat: sym analysis ORC (Lie brack and sym grad)

fedeoli committed:
feat: plot first guesst data

fedeoli committed:
feat: add terminal cost with normalisation + Max Iter Time fix

fedeoli committed:
feat: HPPC modular (no more 6500s tops)

fedeoli committed:
chore: tweak params

fedeoli committed:
feat: improved barrier fcn + terminal norm as a choice

fedeoli committed:
fixed: correct input law

fedeoli committed:
chore: tweak params

fedeoli committed:
feat: hanlde Inf cost function

fedeoli committed:
feat: add delta error in initial coefficients

fedeoli committed:
chore: improve plot interp (first guess)

fedeoli committed:
test: adaptive control - offline model implementation

fedeoli committed:
feat: handle NaN/Inf cost function

fedeoli committed:
chore: quick

fedeoli committed:
fix: adapt to realdata

fedeoli committed:
feat: generate measure and simulation with realdata (no synthetic)

fedeoli committed:
chore: add comment on time selection

fedeoli committed:
chore: edit .gitignore

fedeoli committed:
data: add last data from tushar

fedeoli committed:
chore: change params and move optimise flag

fedeoli committed:
chore: move docs

fedeoli committed:
feat: SIMULINK model and matlab integration

fedeoli committed:
feat: MultiMHE

fedeoli committed:
chore: test simulink wrappers + realdata sim

fedeoli committed:
chore: set params

fedeoli committed:
fix: single parameters init

fedeoli committed:
chore: Ts and Tend only in simulation_

fedeoli committed:
feat: add input scaling in params

fedeoli committed:
feat: add model inconsistemcy for test

fedeoli committed:
fix: first guess - add precision

fedeoli committed:
feat: add better derivative filter (2nd order)

fedeoli committed:
feat: add constrained simplex

fedeoli committed:
fix: separate obs (first guess) from obs_fast

fedeoli committed:
chore: plots

fedeoli committed:
chore: set params

fedeoli committed:
feat: add fminsearch bound

fedeoli committed:
chore: update obsopt

fedeoli committed:
chore: euler integration

fedeoli committed:
feat: annihilator (Tenni)

fedeoli committed:
feat: Zaccarian allocation

fedeoli committed:
chore: gitignore

alessandrotenaglia committed:
data: results

alessandrotenaglia committed:
chore: add subplot

alessandrotenaglia committed:
feat: thinkpad changes

fedeoli committed:
feat: lsim integration (only LTI)

fedeoli committed:
feat: simulation file only for LSIM (some time saved)

fedeoli committed:
fix: add lsim also in propagation after optimization

fedeoli committed:
fix: patternsearch setup

fedeoli committed:
fix: fminsearchcon error

fedeoli committed:
update: model euler compliant with Lsim

fedeoli committed:
feat: nnl cost function (ddz)

fedeoli committed:
fix: ScaleMesh flag in patternsearch

alessandrotenaglia committed:
feat: tenny

alessandrotenaglia committed:
feat: tenny

fedeoli committed:
fix: input dimension in J integration

fedeoli committed:
fix: input dimensions

fedeoli committed:
merge: preparation of obsopt and plot for merging

fedeoli committed:
merge: after IFAC merge with control branch

fedeoli committed:
mearge: files different from obsopt

fedeoli committed:
merge: files different from obsopt

fedeoli committed:
merge: obsopt

fedeoli committed:
Merge branch 'battery' into dev

fedeoli committed:
merge: check both allocation and battery sim

fedeoli committed:
chore: test battery sim

alessandrotenaglia committed:
chore: remove all mat file

alessandrotenaglia committed:
Merge branch 'dev' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into dev

fedeoli committed:
merge: fix before merge with dev (VDP oscillator)

fedeoli committed:
merge: fix conflicts

fedeoli committed:
chore: clean up stuff

fedeoli committed:
Merge branch 'dev' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into dev

fedeoli committed:
test: battery test on fminsearchcon

fedeoli committed:
chore: params and weights

fedeoli committed:
feat: test file on wavelets

fedeoli committed:
feat: test obsopt on wavelets

fedeoli committed:
feat: wavelets on VDP

fedeoli committed:
chore: params

fedeoli committed:
feat: rover model first draft

smatto72 committed:
adding uwb folder

fedeoli committed:
feat: rover params

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

smatto72 committed:
added USAGE and epsilon

smatto72 committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
feat: update rover model

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
chore: tidy up various folder

smatto72 committed:
display parameter

smatto72 committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

smatto72 committed:
no initial condition error

fedeoli committed:
feat: quicker sym gradient and noise on dist

fedeoli committed:
feat: noise on IMU meas

smatto72 committed:
fix: noise with bias

fedeoli committed:
chore: params.noise_mat

smatto72 committed:
basic observer dynamics

fedeoli committed:
feat: HGO + code redesign

fedeoli committed:
feat: Xmas edit (opt setup)

fedeoli committed:
feat: normalise with al trajectories

fedeoli committed:
feat: best improvements in the J for multistart

fedeoli committed:
chore: params

fedeoli committed:
feat: constant output in UWB meas

fedeoli committed:
feat: vectorize - still working

fedeoli committed:
fix: recover after server breakdown

fedeoli committed:
feat: EKF on 6D object

fedeoli committed:
chore: move doc

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
feat: working EKF on Armesto - no bias

fedeoli committed:
feat: EKF on 2D rover

fedeoli committed:
feat: EKF + VolterraLotka model

fedeoli committed:
fix: check the code after a while - battery project

fedeoli committed:
test: same measure file for both EKF and Hyb

fedeoli committed:
fix: measure function

fedeoli committed:
test: periodic sampling + gamma correction jump map

fedeoli committed:
test: adaptive sampling

fedeoli committed:
feat: adaptive sampling working

fedeoli committed:
chore: no constraint on gamma (nonlinear filter)

fedeoli committed:
chore: good params results

fedeoli committed:
chore: adaptive sampling params init in obsopt

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
feat: 3D version of EKF and hyb

fedeoli committed:
chore: nonlinconstraints

fedeoli committed:
feat: hills on z

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
fix: bandpass filter on hyb

fedeoli committed:
Merge branch 'hyb' of github.com:IntelligentSystemsLabUTV/obsopt_pkg into hyb

fedeoli committed:
fix: PD controller + correct Jdist

fedeoli committed:
fix: error derivative

fedeoli committed:
fix: C params constrained

fedeoli committed:
fix: bandpass filter hyb

fedeoli committed:
fix: WaitAllBuffer 2 test

fedeoli committed:
feat: WaitAllBuffer 2 update (full traj target)

fedeoli committed:
test: bandpass testing 3D

fedeoli committed:
test: leaning for realdata

fedeoli committed:
fix: simulink

fedeoli committed:
chore: sim battery fixing

fedeoli committed:
chore: diverged branch

fedeoli committed:
feat: realdata with separate simulink

fedeoli committed:
chore: cleanup

fedeoli committed:
update gitignore

fedeoli committed:
Thesis: sim with Nts

fedeoli committed:
test: battery fminunc

fedeoli committed:
Thesis: test on fminsearchcon

fedeoli committed:
Thesis: testing

fedeoli committed:
Thesis: chore simulink_realdata

fedeoli committed:
Thesis: PE feat on obsotp

tkdesai13 committed:
CALCE dataset trial

- added CALCE dataset and created the files to run on them
- some of these files could be redundant and will be removed later.

tkdesai13 committed:
Initial parameter values and plot modification

-initial parameter values changed
-added the plot_general_calce_to modify the figures (still not finished)

fedeoli committed:
dev: parallel MHE structure for calce

fedeoli committed:
dev: EKF_rover

tkdesai13 committed:
EKF implemented and parameters updated

EKF implemented and parameters updated

tkdesai13 committed:
Data added

Data added

tkdesai13 committed:
updated params

updated params

tkdesai13 committed:
extra plots file

extra plots file

fedeoli committed:
dev: sensitivity analysis

fedeoli committed:
chore: test and clean parallel_calce

fedeoli committed:
chore: update adaptive sampling

fedeoli committed:
chore: update adaptive sampling

fedeoli committed:
chore: update adaptive sampling

fedeoli committed:
chore: update adaptive sampling

fedeoli committed:
chore: update adaptive sampling

fedeoli committed:
dev: update adaptive sampling

fedeoli committed:
dev: tests on CALCE params

fedeoli committed:
chore: fix merge

fedeoli committed:
chore: setup simulations

fedeoli committed:
dev: tests on CALCE params

tkdesai13 committed:
BJDST data added

BJDST data added

tkdesai13 committed:
Create BJDST

tkdesai13 committed:
Delete data/CALCE/INR_18650/BJDST

tkdesai13 committed:
Create 11_11_2015_SP20-2_US06_50SOC.xlsx

US06 data upload

tkdesai13 committed:
Add files via upload

US06 data upload

tkdesai13 committed:
Create 11_09_2015_SP20-2_FUDS_50SOC.xlsx

FUDS data added

tkdesai13 committed:
Add files via upload

tkdesai13 committed:
Create 11_09_2015_SP20-2_FUDS_50SOC.xlsx

BJDST data added

tkdesai13 committed:
Add files via upload

tkdesai13 committed:
Delete data/CALCE/INR_18650/BJDST directory

tkdesai13 committed:
Create 11_13_2015_SP20-2_BJDST_50SOC.xlsx

BJDST data added

tkdesai13 committed:
Add files via upload

tkdesai13 committed:
Merge branch 'battery' of https://github.com/IntelligentSystemsLabUTV/obsopt_pkg into battery

tkdesai13 committed:
data loading updated

- updating data loading as we are not considering soc <0.09 for the current problem
- minor updates in EKF covariances and ininital V1 value.
- Maxiter=1 and terminal_weights are same for all
